internal/apikey/apikey.go

```diff
@@ -141,26 +146,34 @@ func DecodeTKPublicKey(encodedPublicKey string) (*ecdsa.PublicKey, error) {
 	return publicKey, nil
 }
 
-// / Takes a message and returns the proper API stamp
-// / This value should be inserted in a "X-Stamp" header
-func Stamp(message string, apiKey *ApiKey) (string, error) {
-	hash := sha256.Sum256([]byte(message))
+// Signature signs the given message with the given API key.
+// The resulting signature should be added as the "X-Stamp" header of an API request.
+func Signature(message []byte, apiKey *ApiKey) (out []byte, err error) {
```

Comment:
I'm not onboard with renaming this to "Signature", because of the confusion between the cryptography-level concept (ECDSA bytes resulting from ECC signature) and the API-level concept (string attached to a request to prove API key ownership)

"Stamp" is a way for us to hint at the fact that it's _not_ a cryptographic signature. It's a bit more than this (signature encoded and wrapped in a certain way).

Another name we could use: `TkApiSignature`? But given the header is called `X-Stamp`, I'd like to keep the `Stamp` function name to match.